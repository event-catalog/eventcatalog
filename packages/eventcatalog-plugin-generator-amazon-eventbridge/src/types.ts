/* eslint-disable no-unused-vars */
import { Credentials } from '@aws-sdk/types';

// eslint-disable-next-line no-shadow
export enum SchemaTypes {
  JSONSchemaDraft4,
  OpenAPI,
}

export interface PluginOptions {
  credentials: Credentials;
  region: string;
  eventBusName: string;
  registryName: string;
  schemaTypeToRenderToEvent?: SchemaTypes;
  versionEvents?: boolean;
}
