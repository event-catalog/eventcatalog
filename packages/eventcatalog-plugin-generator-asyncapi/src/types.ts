export type AsyncAPIPluginOptions = {
  pathToSpec: string;
  versionEvents?: boolean;
  externalAsyncAPIUrl?: string;
};
