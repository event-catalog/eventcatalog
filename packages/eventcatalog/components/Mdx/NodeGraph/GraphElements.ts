import { ArrowHeadType, XYPosition, Node, Edge } from 'react-flow-renderer';
import getConfig from 'next/config';
import type { Event, Service } from '@eventcatalog/types';

const { publicRuntimeConfig: { basePath = '' } = {} } = getConfig();

const MIN_NODE_WIDTH = 150;
const generateLink = (value, type) => (basePath !== '' ? `${basePath}/${type}/${value}` : `/${type}/${value}`);
const calcWidth = (value) => (value.length * 8 > MIN_NODE_WIDTH ? value.length * 8 : MIN_NODE_WIDTH);

const buildNodeEdge = ({ id, target, source, isAnimated = true }) => ({
  id,
  target,
  source,
  type: 'smoothstep',
  arrowHeadType: ArrowHeadType.ArrowClosed,
  animated: isAnimated,
});

const buildNodeData = ({ label, type, maxWidth }: { label: string; type: 'service' | 'event'; maxWidth?: number }) => {
  const width = calcWidth(label);
  const linkType = type === 'service' ? 'services' : 'events';
  const link = generateLink(label, linkType);
  return { label, link, width, maxWidth };
};

/**
 * Builds a graph for a given event
 * @param {Event} - event
 * @param rootNodeColor - The color of the root node
 * @param isAnimated - whether to animate the graph
 */
export const getEventElements = (
  { name: eventName, producers: eventProducers, consumers: eventConsumers }: Event,
  rootNodeColor = '#2563eb',
  isAnimated = true
) => {
  const position: XYPosition = { x: 0, y: 0 };

  const consumerColor = '#818cf8';
  const producerColor = '#75d7b6';

  const producersNames = eventProducers.map((s) => calcWidth(s));
  const maxProducersWidth = Math.max(...producersNames);
  const consumersNames = eventConsumers.map((s) => calcWidth(s));
  const maxConsumersWidth = Math.max(...consumersNames);

  const eventNameAsNodeID = `ev-${eventName.replace(/ /g, '_')}`;
  const eventNodeWidth = calcWidth(eventName);

  const producers = eventProducers.map((node) => ({ label: node, id: `pr-${node.replace(/ /g, '_')}` }));
  const consumers = eventConsumers.map((node) => ({ label: node, id: `co-${node.replace(/ /g, '_')}` }));

  // Transforms services & event into a graph model
  const producersNodes: Node[] = producers.map(({ label, id }) => {
    const nodeWidth = calcWidth(label);
    const diff = maxProducersWidth - nodeWidth;
    const nodeMaxWidth = diff !== 0 ? nodeWidth - diff : maxProducersWidth;
    return {
      id,
      data: buildNodeData({ label, type: 'service', maxWidth: nodeMaxWidth }),
      style: { border: `2px solid ${producerColor}`, width: nodeWidth },
      type: 'input',
      position,
    };
  });
  const consumersNodes: Node[] = consumers.map(({ id, label }) => {
    const width = calcWidth(label);
    return {
      id,
      data: buildNodeData({ label, type: 'service', maxWidth: maxConsumersWidth }),
      style: { border: `2px solid ${consumerColor}`, width },
      type: 'output',
      position,
    };
  });
  const eventNode: Node = {
    id: eventNameAsNodeID,
    data: buildNodeData({ label: eventName, type: 'event', maxWidth: eventNodeWidth }),
    style: {
      border: `2px solid ${rootNodeColor}`,
      width: eventNodeWidth,
    },
    position,
  };

  // Build connections
  const producersEdges: Edge[] = producers.map(({ id, label }) =>
    buildNodeEdge({ id: `epe-${label.replace(/ /g, '_')}`, source: id, target: eventNameAsNodeID, isAnimated })
  );
  const consumersEdges: Edge[] = consumers.map(({ id, label }) =>
    buildNodeEdge({ id: `ece-${label.replace(/ /g, '_')}`, target: id, source: eventNameAsNodeID, isAnimated })
  );

  // Merge nodes in order
  const elements: (Node | Edge)[] = [...producersNodes, eventNode, ...consumersNodes, ...producersEdges, ...consumersEdges];
  return elements;
};

/**
 * Builds a graph for a given service
 * @param {Service} service
 * @param {string} rootNodeColor of the root node
 * @param isAnimated whether the graph should be animated
 * @returns {string} Mermaid Graph
 */
export const getServiceElements = (
  { publishes, subscribes, name: serviceName }: Service,
  rootNodeColor = '#2563eb',
  isAnimated = true
) => {
  const position: XYPosition = { x: 0, y: 0 };

  const publishColor = '#818cf8';
  const subscribeColor = '#75d7b6';

  const publishesNames = publishes.map((e) => calcWidth(e.name));
  const maxPublishesWidth = Math.max(...publishesNames);
  const subscribesNames = subscribes.map((e) => calcWidth(e.name));
  const maxSubscribesWidth = Math.max(...subscribesNames);

  const serviceNameAsNodeID = `ser-${serviceName.replace(/ /g, '_')}`;

  // Transforms services & event into a graph model
  const publishesNodes: Node[] = publishes.map((node) => {
    const nodeWidth = calcWidth(node.name);
    return {
      id: `pub-${node.name.replace(/ /g, '_')}`,
      data: buildNodeData({ label: node.name, type: 'event', maxWidth: maxPublishesWidth }),
      style: { border: `2px solid ${publishColor}`, width: nodeWidth },
      type: 'output',
      position,
    };
  });
  const subscribesNodes: Node[] = subscribes.map((node) => {
    const nodeWidth = calcWidth(node.name);
    const diff = maxSubscribesWidth - nodeWidth;
    const nodeMaxWidth = diff !== 0 ? nodeWidth - diff : maxSubscribesWidth;
    return {
      id: `sub-${node.name.replace(/ /g, '_')}`,
      data: buildNodeData({ label: node.name, type: 'event', maxWidth: nodeMaxWidth }),
      style: {
        border: `2px solid ${subscribeColor}`,
        width: nodeWidth,
      },
      type: 'input',
      position,
    };
  });

  const serviceNode: Node = {
    id: serviceNameAsNodeID,
    data: buildNodeData({ label: serviceName, type: 'service', maxWidth: calcWidth(serviceName) }),
    style: {
      border: `2px solid ${rootNodeColor}`,
      width: calcWidth(serviceName),
    },
    position,
  };

  // Build connections
  const publishesEdges: Edge[] = publishes.map((node) =>
    buildNodeEdge({
      id: `ecp-${node.name.replace(/ /g, '_')}`,
      source: serviceNameAsNodeID,
      target: `pub-${node.name.replace(/ /g, '_')}`,
      isAnimated,
    })
  );

  const subscribesEdges: Edge[] = subscribes.map((node) =>
    buildNodeEdge({
      id: `esc-${node.name.replace(/ /g, '_')}`,
      target: serviceNameAsNodeID,
      source: `sub-${node.name.replace(/ /g, '_')}`,
      isAnimated,
    })
  );

  // Merge nodes in order
  const elements: (Node | Edge)[] = [...subscribesNodes, serviceNode, ...publishesNodes, ...publishesEdges, ...subscribesEdges];
  return elements;
};
