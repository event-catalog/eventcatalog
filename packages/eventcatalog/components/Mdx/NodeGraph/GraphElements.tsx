import { ArrowHeadType, XYPosition, Node, Edge } from 'react-flow-renderer';
import getConfig from 'next/config';
import type { Event, Service } from '@eventcatalog/types';
import CustomNode from './Node';

const { publicRuntimeConfig: { basePath = '' } = {} } = getConfig();

const MIN_NODE_WIDTH = 150;
type NODE_TYPES = 'service' | 'event';

const generateLink = (value, type, domain?) => `${basePath}/${domain ? `domains/${domain}/` : ''}${type}/${value}`;
const calcWidth = (value) => (value.length * 8 > MIN_NODE_WIDTH ? value.length * 8 : MIN_NODE_WIDTH);

const buildNodeEdge = ({ id, target, source, label, isAnimated = true }): Edge => ({
  id,
  target,
  source,
  type: 'smoothstep',
  arrowHeadType: ArrowHeadType.ArrowClosed,
  animated: isAnimated,
  label,
  labelBgPadding: [8, 4],
  labelBgBorderRadius: 4,
  labelStyle: { fontSize: '6px' },
  labelBgStyle: { fill: 'white', color: '#fff', fillOpacity: 0.5 },
});

const buildNodeData = ({
  name,
  label,
  type,
  maxWidth,
  renderInColumn,
  domain,
}: {
  name: string;
  label: string;
  type: NODE_TYPES;
  maxWidth?: number;
  renderInColumn?: number;
  domain?: string;
}) => {
  const width = calcWidth(label);
  const linkType = type === 'service' ? 'services' : 'events';

  const link = generateLink(name, linkType, domain);
  return { label, link, width, maxWidth, renderInColumn };
};

const getNodeLabel = ({ type, label, includeIcon }: { type: NODE_TYPES; label: any; includeIcon: boolean }) => {
  if (!includeIcon) return label;
  return <CustomNode type={type} label={label} />;
};

/**
 * Builds a graph for a given event
 * @param {Event} - event
 * @param rootNodeColor - The color of the root node
 * @param isAnimated - whether to animate the graph
 */
export const getEventElements = (
  { name: eventName, domain, consumers: eventConsumers = [], producers: eventProducers = [] }: Event,
  rootNodeColor = '#2563eb',
  isAnimated = true,
  includeLabels = false,
  includeNodeIcons = false
) => {
  const position: XYPosition = { x: 0, y: 0 };

  const consumerColor = '#818cf8';
  const producerColor = '#75d7b6';
  const nodeStyles = {
    fontSize: includeNodeIcons ? '8px' : 'auto',
  };

  const producersNames = eventProducers.map((s) => calcWidth(s.name));
  const maxProducersWidth = Math.max(...producersNames);
  const consumersNames = eventConsumers.map((s) => calcWidth(s.name));
  const maxConsumersWidth = Math.max(...consumersNames);

  const eventNameAsNodeID = `ev-${eventName.replace(/ /g, '_')}`;
  const eventNodeWidth = calcWidth(eventName);

  const producers = eventProducers.map((node) => ({
    label: node.name,
    id: `pr-${node.name.replace(/ /g, '_')}`,
    domain: node.domain,
  }));
  const consumers = eventConsumers.map((node) => ({
    label: node.name,
    id: `co-${node.name.replace(/ /g, '_')}`,
    domain: node.domain,
  }));

  // Transforms services & event into a graph model
  const producersNodes: Node[] = producers.map(({ label, id, domain: producerDomain }) => {
    const nodeWidth = calcWidth(label);
    const diff = maxProducersWidth - nodeWidth;
    const nodeMaxWidth = diff !== 0 ? nodeWidth - diff : maxProducersWidth;
    const labelToRender = getNodeLabel({ type: 'service', label, includeIcon: includeNodeIcons });
    return {
      id,
      data: buildNodeData({
        name: label,
        label: labelToRender,
        type: 'service',
        maxWidth: nodeMaxWidth,
        renderInColumn: 1,
        domain: producerDomain,
      }),
      style: { border: `2px solid ${producerColor}`, width: nodeWidth, ...nodeStyles },
      type: 'input',
      position,
    };
  });
  const consumersNodes: Node[] = consumers.map(({ id, label, domain: consumerDomain }) => {
    const width = calcWidth(label);
    const labelToRender = getNodeLabel({ type: 'service', label, includeIcon: includeNodeIcons });
    return {
      id,
      data: buildNodeData({
        name: label,
        label: labelToRender,
        type: 'service',
        maxWidth: maxConsumersWidth,
        renderInColumn: 3,
        domain: consumerDomain,
      }),
      style: { border: `2px solid ${consumerColor}`, width, ...nodeStyles },
      type: 'output',
      position,
    };
  });

  const eventNode: Node = {
    id: eventNameAsNodeID,
    data: buildNodeData({
      name: eventName,
      label: getNodeLabel({ type: 'event', label: eventName, includeIcon: includeNodeIcons }),
      type: 'event',
      maxWidth: eventNodeWidth,
      renderInColumn: 2,
      domain,
    }),
    style: {
      border: `2px solid ${rootNodeColor}`,
      width: eventNodeWidth,
      ...nodeStyles,
    },
    position,
  };

  // Build connections
  const producersEdges: Edge[] = producers.map(({ id, label }) =>
    buildNodeEdge({
      id: `epe-${label.replace(/ /g, '_')}-${eventNameAsNodeID}`,
      source: id,
      target: eventNameAsNodeID,
      isAnimated,
      label: includeLabels ? 'publishes' : '',
    })
  );
  const consumersEdges: Edge[] = consumers.map(({ id, label }) =>
    buildNodeEdge({
      id: `ece-${label.replace(/ /g, '_')}-${eventNameAsNodeID}`,
      target: id,
      source: eventNameAsNodeID,
      isAnimated,
      label: includeLabels ? 'subscribed by' : '',
    })
  );

  // Merge nodes in order
  const elements: (Node | Edge)[] = [...producersNodes, eventNode, ...consumersNodes, ...producersEdges, ...consumersEdges];
  return elements;
};

/**
 * Builds a graph for a given service
 * @param {Service} service
 * @param {string} rootNodeColor of the root node
 * @param isAnimated whether the graph should be animated
 * @returns {string} Mermaid Graph
 */
export const getServiceElements = (
  { publishes, subscribes, name: serviceName, domain }: Service,
  rootNodeColor = '#2563eb',
  isAnimated = true,
  includeEdgeLabels = false,
  includeNodeIcons = false
) => {
  const position: XYPosition = { x: 0, y: 0 };

  const publishColor = '#818cf8';
  const subscribeColor = '#75d7b6';
  const nodeStyles = {
    fontSize: includeNodeIcons ? '8px' : 'auto',
  };

  const publishesNames = publishes.map((e) => calcWidth(e.name));
  const maxPublishesWidth = Math.max(...publishesNames);
  const subscribesNames = subscribes.map((e) => calcWidth(e.name));
  const maxSubscribesWidth = Math.max(...subscribesNames);

  const serviceNameAsNodeID = `ser-${serviceName.replace(/ /g, '_')}`;

  // Transforms services & event into a graph model
  const publishesNodes: Node[] = publishes.map((node) => {
    const nodeWidth = calcWidth(node.name);
    const labelToRender = getNodeLabel({ type: 'event', label: node.name, includeIcon: includeNodeIcons });
    return {
      id: `pub-${node.name.replace(/ /g, '_')}`,
      data: buildNodeData({
        name: node.name,
        label: labelToRender,
        type: 'event',
        maxWidth: maxPublishesWidth,
        renderInColumn: 3,
        domain: node.domain,
      }),
      style: { border: `2px solid ${publishColor}`, width: nodeWidth, ...nodeStyles },
      type: 'output',
      position,
    };
  });
  const subscribesNodes: Node[] = subscribes.map((node) => {
    const nodeWidth = calcWidth(node.name);
    const diff = maxSubscribesWidth - nodeWidth;
    const nodeMaxWidth = diff !== 0 ? nodeWidth - diff : maxSubscribesWidth;
    const labelToRender = getNodeLabel({ type: 'event', label: node.name, includeIcon: includeNodeIcons });
    return {
      id: `sub-${node.name.replace(/ /g, '_')}`,
      data: buildNodeData({
        name: node.name,
        label: labelToRender,
        type: 'event',
        maxWidth: nodeMaxWidth,
        ...nodeStyles,
        renderInColumn: 1,
        domain: node.domain,
      }),
      style: {
        border: `2px solid ${subscribeColor}`,
        width: nodeWidth,
        ...nodeStyles,
      },
      type: 'input',
      position,
    };
  });

  const serviceNode: Node = {
    id: serviceNameAsNodeID,
    data: buildNodeData({
      name: serviceName,
      label: getNodeLabel({ type: 'service', label: serviceName, includeIcon: includeNodeIcons }),
      type: 'service',
      maxWidth: calcWidth(serviceName),
      renderInColumn: 2,
      domain,
    }),
    style: {
      border: `2px solid ${rootNodeColor}`,
      width: calcWidth(serviceName),
      ...nodeStyles,
    },
    position,
  };

  // Build connections
  const publishesEdges: Edge[] = publishes.map((node) =>
    buildNodeEdge({
      id: `ecp-${node.name.replace(/ /g, '_')}`,
      source: serviceNameAsNodeID,
      target: `pub-${node.name.replace(/ /g, '_')}`,
      isAnimated,
      label: includeEdgeLabels ? 'publishes' : '',
    })
  );

  const subscribesEdges: Edge[] = subscribes.map((node) =>
    buildNodeEdge({
      id: `esc-${node.name.replace(/ /g, '_')}`,
      target: serviceNameAsNodeID,
      source: `sub-${node.name.replace(/ /g, '_')}`,
      isAnimated,
      label: includeEdgeLabels ? 'subscribed by' : '',
    })
  );

  // Merge nodes in order
  const elements: (Node | Edge)[] = [...subscribesNodes, serviceNode, ...publishesNodes, ...publishesEdges, ...subscribesEdges];
  return elements;
};
