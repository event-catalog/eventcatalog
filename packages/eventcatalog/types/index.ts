import { MDXRemoteSerializeResult } from 'next-mdx-remote';

export interface MarkdownFile {
  content: string;
  source: MDXRemoteSerializeResult;
  lastModifiedDate: string;
}
