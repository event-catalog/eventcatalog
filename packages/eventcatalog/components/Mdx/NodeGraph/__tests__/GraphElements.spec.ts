import { Event, Service } from '@eventcatalog/types';
import { getEventElements, getServiceElements } from '../GraphElements';

jest.mock('next/config', () => () => ({
  publicRuntimeConfig: {
    basePath: '/docs',
  },
}));

describe('GraphElements', () => {
  describe('getServiceElements', () => {
    it('takes a given Service and returns the ReactFlow elements with relations the events it publishes and consumes', () => {
      const event = { name: 'My Event', version: '0.0.1' };
      const event2 = { name: 'My Event 2', version: '0.0.1' };
      const rootNodeColor = '#2563eb';
      const isAnimated = true;

      const service = {
        id: 'My Service',
        name: 'My Service',
        version: '0.0.1',
        summary: 'Summary',
        publishes: [event],
        subscribes: [event2],
      };

      const result = getServiceElements(service as Service, rootNodeColor, isAnimated);

      expect(result).toMatchSnapshot();
    });

    it('takes a given Service and returns the ReactFlow elements with multiple events & long names', () => {
      const event = { name: 'My Event', version: '0.0.1' };
      const event2 = { name: 'My Event 2', version: '0.0.1' };
      const event3 = { name: 'My Event 3', version: '0.0.1' };
      const event4 = { name: 'very.very.very.very.very.very.very.very.very.very.very.very.long.name.event.4', version: '0.0.1' };
      const event5 = { name: 'very very very very very very very very very very very very long name event 5', version: '0.0.1' };
      const event6 = { name: 'My Event 6', version: '0.0.1' };
      const event7 = { name: 'My Event 7', version: '0.0.1' };
      const event8 = { name: 'My Event 8', version: '0.0.1' };
      const event9 = { name: 'My Event 9', version: '0.0.1' };
      const rootNodeColor = '#2563eb';
      const isAnimated = true;

      const service = {
        id: 'My Service',
        name: 'My Service',
        version: '0.0.1',
        summary: 'Summary',
        publishes: [event, event2, event3, event4],
        subscribes: [event5, event6, event7, event8, event9],
      };

      const result = getServiceElements(service as Service, rootNodeColor, isAnimated);

      expect(result).toMatchSnapshot();
    });
  });

  describe('getEventElements', () => {
    it('takes a given event and returns the ReactFlow elements with relations between the event and its consumers and producers', () => {
      const event = {
        name: 'My Event',
        version: '0.0.1',
        producerNames: ['Service 1'],
        consumerNames: ['Service 2'],
      };
      const rootNodeColor = '#2563eb';
      const isAnimated = true;

      const result = getEventElements(event as Event, rootNodeColor, isAnimated);

      expect(result).toMatchSnapshot();
    });

    it('takes a given event and returns the ReactFlow elements with multiple services & long names', () => {
      const event = {
        name: 'My Event',
        version: '0.0.1',
        producerNames: [
          'Service 1',
          'Service 2',
          'very very very very very very very very very very very very long name Service 3',
        ],
        consumerNames: [
          'Service 4',
          'Service 5',
          'Service 6',
          'Service 7',
          'Service 8',
          'very.very.very.very.very.very.very.very.very.very.very.very.long.name.event.8',
        ],
      };
      const rootNodeColor = '#2563eb';
      const isAnimated = true;

      const result = getEventElements(event as Event, rootNodeColor, isAnimated);

      expect(result).toMatchSnapshot();
    });
  });
});
