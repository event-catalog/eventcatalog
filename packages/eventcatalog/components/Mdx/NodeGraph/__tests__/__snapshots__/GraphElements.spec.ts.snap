// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphElements getEventElements takes a given event and returns the ReactFlow elements with multiple services & long names 1`] = `
Array [
  Object {
    "data": Object {
      "label": "My Event",
      "link": "/docs/events/My Event",
      "maxWidth": 150,
      "renderInColumn": 2,
      "width": 150,
    },
    "id": "ev-My_Event",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #2563eb",
      "fontSize": "auto",
      "width": 150,
    },
  },
]
`;

exports[`GraphElements getEventElements takes a given event and returns the ReactFlow elements with relations between the event and its consumers and producers 1`] = `
Array [
  Object {
    "data": Object {
      "label": "My Event",
      "link": "/docs/events/My Event",
      "maxWidth": 150,
      "renderInColumn": 2,
      "width": 150,
    },
    "id": "ev-My_Event",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #2563eb",
      "fontSize": "auto",
      "width": 150,
    },
  },
]
`;

exports[`GraphElements getServiceElements takes a given Service and returns the ReactFlow elements with multiple events & long names 1`] = `
Array [
  Object {
    "data": Object {
      "label": "very very very very very very very very very very very very long name event 5",
      "link": "/docs/events/very very very very very very very very very very very very long name event 5",
      "maxWidth": 616,
      "renderInColumn": 1,
      "width": 616,
    },
    "id": "sub-very_very_very_very_very_very_very_very_very_very_very_very_long_name_event_5",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #75d7b6",
      "fontSize": "auto",
      "width": 616,
    },
    "type": "input",
  },
  Object {
    "data": Object {
      "label": "My Event 6",
      "link": "/docs/events/My Event 6",
      "maxWidth": -316,
      "renderInColumn": 1,
      "width": 150,
    },
    "id": "sub-My_Event_6",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #75d7b6",
      "fontSize": "auto",
      "width": 150,
    },
    "type": "input",
  },
  Object {
    "data": Object {
      "label": "My Event 7",
      "link": "/docs/events/My Event 7",
      "maxWidth": -316,
      "renderInColumn": 1,
      "width": 150,
    },
    "id": "sub-My_Event_7",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #75d7b6",
      "fontSize": "auto",
      "width": 150,
    },
    "type": "input",
  },
  Object {
    "data": Object {
      "label": "My Event 8",
      "link": "/docs/events/My Event 8",
      "maxWidth": -316,
      "renderInColumn": 1,
      "width": 150,
    },
    "id": "sub-My_Event_8",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #75d7b6",
      "fontSize": "auto",
      "width": 150,
    },
    "type": "input",
  },
  Object {
    "data": Object {
      "label": "My Event 9",
      "link": "/docs/events/My Event 9",
      "maxWidth": -316,
      "renderInColumn": 1,
      "width": 150,
    },
    "id": "sub-My_Event_9",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #75d7b6",
      "fontSize": "auto",
      "width": 150,
    },
    "type": "input",
  },
  Object {
    "data": Object {
      "label": "My Service",
      "link": "/docs/services/My Service",
      "maxWidth": 150,
      "renderInColumn": 2,
      "width": 150,
    },
    "id": "ser-My_Service",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #2563eb",
      "fontSize": "auto",
      "width": 150,
    },
  },
  Object {
    "data": Object {
      "label": "My Event",
      "link": "/docs/events/My Event",
      "maxWidth": 616,
      "renderInColumn": 3,
      "width": 150,
    },
    "id": "pub-My_Event",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #818cf8",
      "fontSize": "auto",
      "width": 150,
    },
    "type": "output",
  },
  Object {
    "data": Object {
      "label": "My Event 2",
      "link": "/docs/events/My Event 2",
      "maxWidth": 616,
      "renderInColumn": 3,
      "width": 150,
    },
    "id": "pub-My_Event_2",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #818cf8",
      "fontSize": "auto",
      "width": 150,
    },
    "type": "output",
  },
  Object {
    "data": Object {
      "label": "My Event 3",
      "link": "/docs/events/My Event 3",
      "maxWidth": 616,
      "renderInColumn": 3,
      "width": 150,
    },
    "id": "pub-My_Event_3",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #818cf8",
      "fontSize": "auto",
      "width": 150,
    },
    "type": "output",
  },
  Object {
    "data": Object {
      "label": "very.very.very.very.very.very.very.very.very.very.very.very.long.name.event.4",
      "link": "/docs/events/very.very.very.very.very.very.very.very.very.very.very.very.long.name.event.4",
      "maxWidth": 616,
      "renderInColumn": 3,
      "width": 616,
    },
    "id": "pub-very.very.very.very.very.very.very.very.very.very.very.very.long.name.event.4",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #818cf8",
      "fontSize": "auto",
      "width": 616,
    },
    "type": "output",
  },
  Object {
    "animated": true,
    "arrowHeadType": "arrowclosed",
    "id": "ecp-My_Event",
    "label": "",
    "labelBgBorderRadius": 4,
    "labelBgPadding": Array [
      8,
      4,
    ],
    "labelBgStyle": Object {
      "color": "#fff",
      "fill": "white",
      "fillOpacity": 0.5,
    },
    "labelStyle": Object {
      "fontSize": "6px",
    },
    "source": "ser-My_Service",
    "target": "pub-My_Event",
    "type": "smoothstep",
  },
  Object {
    "animated": true,
    "arrowHeadType": "arrowclosed",
    "id": "ecp-My_Event_2",
    "label": "",
    "labelBgBorderRadius": 4,
    "labelBgPadding": Array [
      8,
      4,
    ],
    "labelBgStyle": Object {
      "color": "#fff",
      "fill": "white",
      "fillOpacity": 0.5,
    },
    "labelStyle": Object {
      "fontSize": "6px",
    },
    "source": "ser-My_Service",
    "target": "pub-My_Event_2",
    "type": "smoothstep",
  },
  Object {
    "animated": true,
    "arrowHeadType": "arrowclosed",
    "id": "ecp-My_Event_3",
    "label": "",
    "labelBgBorderRadius": 4,
    "labelBgPadding": Array [
      8,
      4,
    ],
    "labelBgStyle": Object {
      "color": "#fff",
      "fill": "white",
      "fillOpacity": 0.5,
    },
    "labelStyle": Object {
      "fontSize": "6px",
    },
    "source": "ser-My_Service",
    "target": "pub-My_Event_3",
    "type": "smoothstep",
  },
  Object {
    "animated": true,
    "arrowHeadType": "arrowclosed",
    "id": "ecp-very.very.very.very.very.very.very.very.very.very.very.very.long.name.event.4",
    "label": "",
    "labelBgBorderRadius": 4,
    "labelBgPadding": Array [
      8,
      4,
    ],
    "labelBgStyle": Object {
      "color": "#fff",
      "fill": "white",
      "fillOpacity": 0.5,
    },
    "labelStyle": Object {
      "fontSize": "6px",
    },
    "source": "ser-My_Service",
    "target": "pub-very.very.very.very.very.very.very.very.very.very.very.very.long.name.event.4",
    "type": "smoothstep",
  },
  Object {
    "animated": true,
    "arrowHeadType": "arrowclosed",
    "id": "esc-very_very_very_very_very_very_very_very_very_very_very_very_long_name_event_5",
    "label": "",
    "labelBgBorderRadius": 4,
    "labelBgPadding": Array [
      8,
      4,
    ],
    "labelBgStyle": Object {
      "color": "#fff",
      "fill": "white",
      "fillOpacity": 0.5,
    },
    "labelStyle": Object {
      "fontSize": "6px",
    },
    "source": "sub-very_very_very_very_very_very_very_very_very_very_very_very_long_name_event_5",
    "target": "ser-My_Service",
    "type": "smoothstep",
  },
  Object {
    "animated": true,
    "arrowHeadType": "arrowclosed",
    "id": "esc-My_Event_6",
    "label": "",
    "labelBgBorderRadius": 4,
    "labelBgPadding": Array [
      8,
      4,
    ],
    "labelBgStyle": Object {
      "color": "#fff",
      "fill": "white",
      "fillOpacity": 0.5,
    },
    "labelStyle": Object {
      "fontSize": "6px",
    },
    "source": "sub-My_Event_6",
    "target": "ser-My_Service",
    "type": "smoothstep",
  },
  Object {
    "animated": true,
    "arrowHeadType": "arrowclosed",
    "id": "esc-My_Event_7",
    "label": "",
    "labelBgBorderRadius": 4,
    "labelBgPadding": Array [
      8,
      4,
    ],
    "labelBgStyle": Object {
      "color": "#fff",
      "fill": "white",
      "fillOpacity": 0.5,
    },
    "labelStyle": Object {
      "fontSize": "6px",
    },
    "source": "sub-My_Event_7",
    "target": "ser-My_Service",
    "type": "smoothstep",
  },
  Object {
    "animated": true,
    "arrowHeadType": "arrowclosed",
    "id": "esc-My_Event_8",
    "label": "",
    "labelBgBorderRadius": 4,
    "labelBgPadding": Array [
      8,
      4,
    ],
    "labelBgStyle": Object {
      "color": "#fff",
      "fill": "white",
      "fillOpacity": 0.5,
    },
    "labelStyle": Object {
      "fontSize": "6px",
    },
    "source": "sub-My_Event_8",
    "target": "ser-My_Service",
    "type": "smoothstep",
  },
  Object {
    "animated": true,
    "arrowHeadType": "arrowclosed",
    "id": "esc-My_Event_9",
    "label": "",
    "labelBgBorderRadius": 4,
    "labelBgPadding": Array [
      8,
      4,
    ],
    "labelBgStyle": Object {
      "color": "#fff",
      "fill": "white",
      "fillOpacity": 0.5,
    },
    "labelStyle": Object {
      "fontSize": "6px",
    },
    "source": "sub-My_Event_9",
    "target": "ser-My_Service",
    "type": "smoothstep",
  },
]
`;

exports[`GraphElements getServiceElements takes a given Service and returns the ReactFlow elements with relations the events it publishes and consumes 1`] = `
Array [
  Object {
    "data": Object {
      "label": "My Event 2",
      "link": "/docs/events/My Event 2",
      "maxWidth": 150,
      "renderInColumn": 1,
      "width": 150,
    },
    "id": "sub-My_Event_2",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #75d7b6",
      "fontSize": "auto",
      "width": 150,
    },
    "type": "input",
  },
  Object {
    "data": Object {
      "label": "My Service",
      "link": "/docs/services/My Service",
      "maxWidth": 150,
      "renderInColumn": 2,
      "width": 150,
    },
    "id": "ser-My_Service",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #2563eb",
      "fontSize": "auto",
      "width": 150,
    },
  },
  Object {
    "data": Object {
      "label": "My Event",
      "link": "/docs/events/My Event",
      "maxWidth": 150,
      "renderInColumn": 3,
      "width": 150,
    },
    "id": "pub-My_Event",
    "position": Object {
      "x": 0,
      "y": 0,
    },
    "style": Object {
      "border": "2px solid #818cf8",
      "fontSize": "auto",
      "width": 150,
    },
    "type": "output",
  },
  Object {
    "animated": true,
    "arrowHeadType": "arrowclosed",
    "id": "ecp-My_Event",
    "label": "",
    "labelBgBorderRadius": 4,
    "labelBgPadding": Array [
      8,
      4,
    ],
    "labelBgStyle": Object {
      "color": "#fff",
      "fill": "white",
      "fillOpacity": 0.5,
    },
    "labelStyle": Object {
      "fontSize": "6px",
    },
    "source": "ser-My_Service",
    "target": "pub-My_Event",
    "type": "smoothstep",
  },
  Object {
    "animated": true,
    "arrowHeadType": "arrowclosed",
    "id": "esc-My_Event_2",
    "label": "",
    "labelBgBorderRadius": 4,
    "labelBgPadding": Array [
      8,
      4,
    ],
    "labelBgStyle": Object {
      "color": "#fff",
      "fill": "white",
      "fillOpacity": 0.5,
    },
    "labelStyle": Object {
      "fontSize": "6px",
    },
    "source": "sub-My_Event_2",
    "target": "ser-My_Service",
    "type": "smoothstep",
  },
]
`;
